src/main/java/fmartin1/controller/PokedexController.java

```diff
@@ -0,0 +1,79 @@
+package fmartin1.controller;
+
+
+import fmartin1.model.pokemon.Pokemon;
+import fmartin1.model.pokemon.PokemonComparator;
+import fmartin1.service.PokedexService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.stream.Collectors;
+
+@SuppressWarnings("unused")
+@Controller
+@RequestMapping("pokemon")
+public class PokedexController {
```

Comment:
I was thinking about that this controller should not have any logic, please move all filtering, mapping and collecting to service and create methods accordingly

Reply:
In the end the responsibility to filter, sort and so belongs to PokemonService, the controller should only be in charge of return these values in the correct format